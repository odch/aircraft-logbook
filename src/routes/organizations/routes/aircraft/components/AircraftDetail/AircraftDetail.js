import React from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage } from 'react-intl'
import { Redirect } from 'react-router-dom'
import withStyles from '@material-ui/core/styles/withStyles'
import Typography from '@material-ui/core/Typography'
import Button from '@material-ui/core/Button'
import isLoaded from '../../../../../../util/isLoaded'
import {
  organization as organizationShape,
  aircraft as aircraftShape,
  flight as flightShape
} from '../../../../../../shapes'
import LoadingIcon from '../../../../../../components/LoadingIcon'
import FlightList from '../FlightList'
import FlightCreateDialog from '../../containers/FlightCreateDialogContainer'

const styles = theme => ({
  container: {
    padding: '1em',
    [theme.breakpoints.up(1000 + theme.spacing(3 * 2))]: {
      width: 1000,
      marginLeft: 'auto',
      marginRight: 'auto'
    }
  }
})

class AircraftDetail extends React.Component {
  handleCreateClick = () => {
    const { organization, aircraft } = this.props
    this.props.openCreateFlightDialog()
    this.props.initCreateFlightDialog(organization.id, aircraft.id)
  }

  componentDidMount() {
    const {
      organization,
      aircraft,
      flightsPagination,
      fetchAircrafts,
      fetchMembers,
      fetchFlights
    } = this.props

    if (organization) {
      fetchAircrafts(organization.id)
      fetchMembers(organization.id)

      if (aircraft) {
        fetchFlights(
          organization.id,
          aircraft.id,
          flightsPagination.page,
          flightsPagination.rowsPerPage
        )
      }
    }
  }

  componentDidUpdate(prevProps) {
    const {
      organization,
      aircraft,
      flightsPagination,
      fetchAircrafts,
      fetchMembers,
      fetchFlights
    } = this.props

    if (
      organization &&
      (!prevProps.organization || prevProps.organization.id !== organization.id)
    ) {
      fetchAircrafts(organization.id)
      fetchMembers(organization.id)
    }

    if (
      organization &&
      aircraft &&
      (!prevProps.aircraft ||
        prevProps.aircraft.id !== aircraft.id ||
        prevProps.flightsPagination.page !== flightsPagination.page ||
        prevProps.flightsPagination.rowsPerPage !==
          flightsPagination.rowsPerPage)
    ) {
      fetchFlights(
        organization.id,
        aircraft.id,
        flightsPagination.page,
        flightsPagination.rowsPerPage
      )
    }
  }

  render() {
    const {
      organization,
      aircraft,
      flights,
      flightDeleteDialog,
      flightsPagination,
      classes,
      createFlightDialogOpen,
      openDeleteFlightDialog,
      closeDeleteFlightDialog,
      deleteFlight,
      setFlightsPage
    } = this.props

    if (!isLoaded(organization) || !isLoaded(aircraft) || !isLoaded(flights)) {
      return <LoadingIcon />
    }

    if (organization === null) {
      return <Redirect to="/" />
    }

    if (aircraft === null) {
      return <Redirect to={`/organizations/${organization.id}`} />
    }

    return (
      <div className={classes.container}>
        <Typography variant="h4" gutterBottom>
          {aircraft.registration}
        </Typography>
        <Button
          variant="contained"
          color="primary"
          onClick={this.handleCreateClick}
        >
          <FormattedMessage id="aircraftdetail.createflight" />
        </Button>
        {flights.length > 0 ? (
          <FlightList
            organization={organization}
            aircraft={aircraft}
            flights={flights}
            flightDeleteDialog={flightDeleteDialog}
            pagination={flightsPagination}
            openFlightDeleteDialog={openDeleteFlightDialog}
            closeFlightDeleteDialog={closeDeleteFlightDialog}
            deleteFlight={deleteFlight}
            setFlightsPage={setFlightsPage}
          />
        ) : (
          <Typography paragraph>
            <FormattedMessage id="aircraftdetail.noflights" />
          </Typography>
        )}
        {createFlightDialogOpen && (
          <FlightCreateDialog
            organizationId={organization.id}
            aircraftId={aircraft.id}
          />
        )}
      </div>
    )
  }
}

AircraftDetail.propTypes = {
  organization: organizationShape,
  aircraft: aircraftShape,
  flights: PropTypes.arrayOf(flightShape),
  flightsPagination: PropTypes.shape({
    rowsCount: PropTypes.number.isRequired,
    page: PropTypes.number.isRequired,
    rowsPerPage: PropTypes.number.isRequired
  }),
  classes: PropTypes.object.isRequired,
  createFlightDialogOpen: PropTypes.bool.isRequired,
  flightDeleteDialog: PropTypes.shape({
    open: PropTypes.bool,
    submitted: PropTypes.bool,
    flight: flightShape
  }).isRequired,
  fetchAircrafts: PropTypes.func.isRequired,
  fetchMembers: PropTypes.func.isRequired,
  fetchFlights: PropTypes.func.isRequired,
  openCreateFlightDialog: PropTypes.func.isRequired,
  initCreateFlightDialog: PropTypes.func.isRequired,
  openDeleteFlightDialog: PropTypes.func.isRequired,
  closeDeleteFlightDialog: PropTypes.func.isRequired,
  deleteFlight: PropTypes.func.isRequired,
  setFlightsPage: PropTypes.func.isRequired
}

export default withStyles(styles)(AircraftDetail)
