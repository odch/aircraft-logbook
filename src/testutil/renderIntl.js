import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
