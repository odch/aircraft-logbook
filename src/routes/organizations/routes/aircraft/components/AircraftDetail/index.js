import AircraftDetail from './AircraftDetail'

export default AircraftDetail
