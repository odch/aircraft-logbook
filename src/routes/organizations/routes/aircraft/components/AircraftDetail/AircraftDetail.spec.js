import React from 'react'
import { BrowserRouter as Router, Switch, Route } from 'react-router-dom'
import renderIntl from '../../../../../../testutil/renderIntl'
import AircraftDetail from './AircraftDetail'

const StartPage = () => <div>Start Page</div>

describe('routes', () => {
  describe('organizations', () => {
    describe('routes', () => {
      describe('aircraft', () => {
        describe('components', () => {
          describe('AircraftDetail', () => {
            it('renders loading icon if organization not loaded', () => {
              const tree = renderIntl(
                <AircraftDetail
                  organization={undefined}
                  createFlightDialogOpen={false}
                  flightDeleteDialog={{ open: false }}
                  fetchAircrafts={() => {}}
                  fetchFlights={() => {}}
                  fetchMembers={() => {}}
                  fetchAerodromes={() => {}}
                  openCreateFlightDialog={() => {}}
                  initCreateFlightDialog={() => {}}
                  openDeleteFlightDialog={() => {}}
                  closeDeleteFlightDialog={() => {}}
                  deleteFlight={() => {}}
                  setFlightsPage={() => {}}
                />
              ).toJSON()
              expect(tree).toMatchSnapshot()
            })

            it('renders loading icon if aircraft not loaded', () => {
              const tree = renderIntl(
                <AircraftDetail
                  organization={{ id: 'my_org' }}
                  aircraft={undefined}
                  createFlightDialogOpen={false}
                  flightDeleteDialog={{ open: false }}
                  fetchAircrafts={() => {}}
                  fetchFlights={() => {}}
                  fetchMembers={() => {}}
                  fetchAerodromes={() => {}}
                  openCreateFlightDialog={() => {}}
                  initCreateFlightDialog={() => {}}
                  openDeleteFlightDialog={() => {}}
                  closeDeleteFlightDialog={() => {}}
                  deleteFlight={() => {}}
                  setFlightsPage={() => {}}
                />
              ).toJSON()
              expect(tree).toMatchSnapshot()
            })

            it('redirects to start page if organization was not found', () => {
              const tree = renderIntl(
                <Router>
                  <Switch>
                    <Route exact path="/" component={StartPage} />
                    <AircraftDetail
                      organization={null}
                      createFlightDialogOpen={false}
                      flightDeleteDialog={{ open: false }}
                      fetchAircrafts={() => {}}
                      fetchFlights={() => {}}
                      fetchMembers={() => {}}
                      fetchAerodromes={() => {}}
                      openCreateFlightDialog={() => {}}
                      initCreateFlightDialog={() => {}}
                      openDeleteFlightDialog={() => {}}
                      closeDeleteFlightDialog={() => {}}
                      deleteFlight={() => {}}
                      setFlightsPage={() => {}}
                    />
                  </Switch>
                </Router>
              ).toJSON()
              expect(tree).toMatchSnapshot()
            })

            it('renders loading icon if flights not loaded', () => {
              const tree = renderIntl(
                <Router>
                  <AircraftDetail
                    organization={{ id: 'my_org' }}
                    aircraft={{
                      id: 'o7flC7jw8jmkOfWo8oyA',
                      registration: 'HBKFW'
                    }}
                    flights={undefined}
                    flightsPagination={{
                      page: 0,
                      rowsPerPage: 10,
                      rowsCount: 45
                    }}
                    createFlightDialogOpen={false}
                    flightDeleteDialog={{ open: false }}
                    fetchAircrafts={() => {}}
                    fetchFlights={() => {}}
                    fetchMembers={() => {}}
                    fetchAerodromes={() => {}}
                    openCreateFlightDialog={() => {}}
                    initCreateFlightDialog={() => {}}
                    openDeleteFlightDialog={() => {}}
                    closeDeleteFlightDialog={() => {}}
                    deleteFlight={() => {}}
                    setFlightsPage={() => {}}
                  />
                </Router>
              ).toJSON()
              expect(tree).toMatchSnapshot()
            })

            it('renders aircraft detail when everything is loaded', () => {
              const tree = renderIntl(
                <Router>
                  <AircraftDetail
                    organization={{ id: 'my_org' }}
                    aircraft={{
                      id: 'o7flC7jw8jmkOfWo8oyA',
                      registration: 'HBKFW'
                    }}
                    flights={[
                      {
                        id: 'sStfyLd2XArT7oUZPFDn',
                        departureAerodrome: {
                          identification: 'LSZT',
                          timezone: 'Europe/Zurich'
                        },
                        destinationAerodrome: {
                          identification: 'LSZT',
                          timezone: 'Europe/Zurich'
                        },
                        blockOffTime: {
                          toDate: () => Date.parse('2018-11-20 10:00 GMT+0100')
                        },
                        blockOnTime: {
                          toDate: () => Date.parse('2018-11-20 11:10 GMT+0100')
                        },
                        takeOffTime: {
                          toDate: () => Date.parse('2018-11-20 10:10 GMT+0100')
                        },
                        landingTime: {
                          toDate: () => Date.parse('2018-11-20 11:00 GMT+0100')
                        },
                        pilot: {
                          firstname: 'Max',
                          lastname: 'Muster'
                        }
                      }
                    ]}
                    flightsPagination={{
                      page: 0,
                      rowsPerPage: 10,
                      rowsCount: 45
                    }}
                    createFlightDialogOpen={false}
                    flightDeleteDialog={{ open: false }}
                    fetchAircrafts={() => {}}
                    fetchFlights={() => {}}
                    fetchMembers={() => {}}
                    fetchAerodromes={() => {}}
                    openCreateFlightDialog={() => {}}
                    initCreateFlightDialog={() => {}}
                    openDeleteFlightDialog={() => {}}
                    closeDeleteFlightDialog={() => {}}
                    deleteFlight={() => {}}
                    setFlightsPage={() => {}}
                  />
                </Router>
              ).toJSON()
              expect(tree).toMatchSnapshot()
            })

            it('calls fetchAircrafts when mounted with organization', () => {
              const fetchAircrafts = jest.fn()

              renderIntl(
                <AircraftDetail
                  organization={{ id: 'my_org' }}
                  aircraft={undefined}
                  createFlightDialogOpen={false}
                  flightDeleteDialog={{ open: false }}
                  fetchAircrafts={fetchAircrafts}
                  fetchFlights={() => {}}
                  fetchMembers={() => {}}
                  fetchAerodromes={() => {}}
                  openCreateFlightDialog={() => {}}
                  initCreateFlightDialog={() => {}}
                  openDeleteFlightDialog={() => {}}
                  closeDeleteFlightDialog={() => {}}
                  deleteFlight={() => {}}
                  setFlightsPage={() => {}}
                />
              )

              expect(fetchAircrafts).toBeCalledWith('my_org')
            })

            it('calls fetchFlights when mounted with organization and aircraft', () => {
              const fetchFlights = jest.fn()

              renderIntl(
                <AircraftDetail
                  organization={{ id: 'my_org' }}
                  aircraft={{
                    id: 'o7flC7jw8jmkOfWo8oyA',
                    registration: 'HBKFW'
                  }}
                  flightsPagination={{
                    page: 0,
                    rowsPerPage: 10,
                    rowsCount: 45
                  }}
                  createFlightDialogOpen={false}
                  flightDeleteDialog={{ open: false }}
                  fetchAircrafts={() => {}}
                  fetchFlights={fetchFlights}
                  fetchMembers={() => {}}
                  fetchAerodromes={() => {}}
                  openCreateFlightDialog={() => {}}
                  initCreateFlightDialog={() => {}}
                  openDeleteFlightDialog={() => {}}
                  closeDeleteFlightDialog={() => {}}
                  deleteFlight={() => {}}
                  setFlightsPage={() => {}}
                />
              )

              expect(fetchFlights).toBeCalledWith(
                'my_org',
                'o7flC7jw8jmkOfWo8oyA',
                0,
                10
              )
            })

            it('does not call fetchAircrafts when mounted without organization', () => {
              const fetchAircrafts = jest.fn()

              renderIntl(
                <AircraftDetail
                  organization={undefined}
                  aircraft={undefined}
                  createFlightDialogOpen={false}
                  flightDeleteDialog={{ open: false }}
                  fetchAircrafts={fetchAircrafts}
                  fetchFlights={() => {}}
                  fetchMembers={() => {}}
                  fetchAerodromes={() => {}}
                  openCreateFlightDialog={() => {}}
                  initCreateFlightDialog={() => {}}
                  openDeleteFlightDialog={() => {}}
                  closeDeleteFlightDialog={() => {}}
                  deleteFlight={() => {}}
                  setFlightsPage={() => {}}
                />
              )

              expect(fetchAircrafts).not.toBeCalled()
            })

            it('does not call fetchFlights when mounted without organization', () => {
              const fetchAircrafts = jest.fn()
              const fetchFlights = jest.fn()

              renderIntl(
                <AircraftDetail
                  organization={undefined}
                  aircraft={undefined}
                  createFlightDialogOpen={false}
                  flightDeleteDialog={{ open: false }}
                  fetchAircrafts={fetchAircrafts}
                  fetchFlights={fetchFlights}
                  fetchMembers={() => {}}
                  fetchAerodromes={() => {}}
                  openCreateFlightDialog={() => {}}
                  initCreateFlightDialog={() => {}}
                  openDeleteFlightDialog={() => {}}
                  closeDeleteFlightDialog={() => {}}
                  deleteFlight={() => {}}
                  setFlightsPage={() => {}}
                />
              )

              expect(fetchAircrafts).not.toBeCalled()
            })

            it('calls fetchAircrafts when updated with new organization', () => {
              const fetchAircrafts = jest.fn()

              const renderer = renderIntl(
                <AircraftDetail
                  organization={undefined}
                  aircraft={undefined}
                  createFlightDialogOpen={false}
                  flightDeleteDialog={{ open: false }}
                  fetchAircrafts={fetchAircrafts}
                  fetchFlights={() => {}}
                  fetchMembers={() => {}}
                  fetchAerodromes={() => {}}
                  openCreateFlightDialog={() => {}}
                  initCreateFlightDialog={() => {}}
                  openDeleteFlightDialog={() => {}}
                  closeDeleteFlightDialog={() => {}}
                  deleteFlight={() => {}}
                  setFlightsPage={() => {}}
                />
              )

              expect(fetchAircrafts).not.toBeCalled()

              renderer.update(
                <AircraftDetail
                  organization={{ id: 'my_org' }}
                  aircraft={undefined}
                  createFlightDialogOpen={false}
                  flightDeleteDialog={{ open: false }}
                  fetchAircrafts={fetchAircrafts}
                  fetchFlights={() => {}}
                  fetchMembers={() => {}}
                  fetchAerodromes={() => {}}
                  openCreateFlightDialog={() => {}}
                  initCreateFlightDialog={() => {}}
                  openDeleteFlightDialog={() => {}}
                  closeDeleteFlightDialog={() => {}}
                  deleteFlight={() => {}}
                  setFlightsPage={() => {}}
                />
              )

              expect(fetchAircrafts).toBeCalledWith('my_org')
            })

            it('calls fetchFlights when updated with new aircraft', () => {
              const fetchFlights = jest.fn()

              const renderer = renderIntl(
                <AircraftDetail
                  organization={{ id: 'my_org' }}
                  aircraft={undefined}
                  flightsPagination={{
                    page: 0,
                    rowsPerPage: 10,
                    rowsCount: 45
                  }}
                  createFlightDialogOpen={false}
                  flightDeleteDialog={{ open: false }}
                  fetchAircrafts={() => {}}
                  fetchFlights={fetchFlights}
                  fetchMembers={() => {}}
                  fetchAerodromes={() => {}}
                  openCreateFlightDialog={() => {}}
                  initCreateFlightDialog={() => {}}
                  openDeleteFlightDialog={() => {}}
                  closeDeleteFlightDialog={() => {}}
                  deleteFlight={() => {}}
                  setFlightsPage={() => {}}
                />
              )

              expect(fetchFlights).not.toBeCalled()

              renderer.update(
                <AircraftDetail
                  organization={{ id: 'my_org' }}
                  aircraft={{
                    id: 'o7flC7jw8jmkOfWo8oyA',
                    registration: 'HBKFW'
                  }}
                  flightsPagination={{
                    page: 0,
                    rowsPerPage: 10,
                    rowsCount: 45
                  }}
                  createFlightDialogOpen={false}
                  flightDeleteDialog={{ open: false }}
                  fetchAircrafts={() => {}}
                  fetchFlights={fetchFlights}
                  fetchMembers={() => {}}
                  fetchAerodromes={() => {}}
                  openCreateFlightDialog={() => {}}
                  initCreateFlightDialog={() => {}}
                  openDeleteFlightDialog={() => {}}
                  closeDeleteFlightDialog={() => {}}
                  deleteFlight={() => {}}
                  setFlightsPage={() => {}}
                />
              )

              expect(fetchFlights).toBeCalledWith(
                'my_org',
                'o7flC7jw8jmkOfWo8oyA',
                0,
                10
              )
            })
          })
        })
      })
    })
  })
})
