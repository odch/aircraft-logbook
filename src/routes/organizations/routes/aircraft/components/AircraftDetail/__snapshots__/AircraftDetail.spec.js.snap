// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes aircraft components AircraftDetail redirects to start page if organization was not found 1`] = `
<div>
  Start Page
</div>
`;

exports[`routes organizations routes aircraft components AircraftDetail renders aircraft detail when everything is loaded 1`] = `
<div
  className="AircraftDetail-container-1"
>
  <h4
    className="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom"
  >
    HBKFW
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root AircraftDetail-settingsButton-3 MuiButton-text MuiButton-textPrimary"
      href="/organizations/my_org/aircrafts/o7flC7jw8jmkOfWo8oyA/settings"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        Einstellungen bearbeiten
      </span>
    </a>
  </h4>
  <h5
    className="MuiTypography-root AircraftDetail-sectionHeading-2 MuiTypography-h5 MuiTypography-gutterBottom"
  >
    Offene Techlog-Einträge
  </h5>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Eintrag erfassen
    </span>
  </button>
  <div
    className="Techlog-container-69"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1 MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-rounded"
    >
      <div
        aria-disabled={false}
        aria-expanded={false}
        className="MuiButtonBase-root MuiExpansionPanelSummary-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiExpansionPanelSummary-content"
        >
          <div
            className="Techlog-entryHeading-70"
          >
            <p
              className="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph"
            >
              Schraube am Bugfahrwerk locker
              <br />
            </p>
            <span
              className="MuiTypography-root EntryStatus-status-115 EntryStatus-severe-116 MuiTypography-body1"
            >
              Defect (not airworthy)
            </span>
          </div>
          <p
            className="MuiTypography-root Techlog-entrySecondaryHeading-71 MuiTypography-body1"
          >
            Hans
             
            Meier
            , 
            25.06.2020
             
            09:54
          </p>
        </div>
        <div
          aria-disabled={false}
          aria-hidden={true}
          className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={-1}
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        className="MuiCollapse-container MuiCollapse-hidden"
        style={
          Object {
            "minHeight": "0px",
          }
        }
      >
        <div
          className="MuiCollapse-wrapper"
        >
          <div
            className="MuiCollapse-wrapperInner"
          >
            <div
              role="region"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
    href="/organizations/my_org/aircrafts/o7flC7jw8jmkOfWo8oyA/techlog"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <span
      className="MuiButton-label"
    >
      Alle anzeigen
    </span>
  </a>
  <h5
    className="MuiTypography-root AircraftDetail-sectionHeading-2 MuiTypography-h5 MuiTypography-gutterBottom"
  >
    Neueste Flüge
  </h5>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary"
    disabled={false}
    onBlur={[Function]}
    onClick={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      Flug erfassen
    </span>
  </button>
  <div
    className="FlightList-container-144"
  >
    <div
      className="MuiPaper-root MuiPaper-elevation1 MuiExpansionPanel-root MuiExpansionPanel-rounded MuiPaper-rounded"
      data-id="sStfyLd2XArT7oUZPFDn"
    >
      <div
        aria-disabled={false}
        aria-expanded={false}
        className="MuiButtonBase-root MuiExpansionPanelSummary-root"
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="button"
        tabIndex={0}
      >
        <div
          className="MuiExpansionPanelSummary-content"
        >
          <div
            className="FlightList-flightHeading-145"
          >
            <p
              className="MuiTypography-root MuiTypography-body1"
            >
              Von 
              <span
                className="FlightList-bold-147"
              >
                LSZT
              </span>
               nach 
              <span
                className="FlightList-bold-147"
              >
                LSZT
              </span>
               — Max Muster
            </p>
          </div>
          <p
            className="MuiTypography-root FlightList-flightSecondaryHeading-146 MuiTypography-body1"
          >
            20.11.2018
            ,
             
            10:00
            -
            11:10
          </p>
        </div>
        <div
          aria-disabled={false}
          aria-hidden={true}
          className="MuiButtonBase-root MuiIconButton-root MuiExpansionPanelSummary-expandIcon MuiIconButton-edgeEnd"
          onBlur={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          role="button"
          tabIndex={-1}
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
              />
            </svg>
          </span>
        </div>
      </div>
      <div
        className="MuiCollapse-container MuiCollapse-hidden"
        style={
          Object {
            "minHeight": "0px",
          }
        }
      >
        <div
          className="MuiCollapse-wrapper"
        >
          <div
            className="MuiCollapse-wrapperInner"
          >
            <div
              role="region"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
  <a
    aria-disabled={false}
    className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textPrimary"
    href="/organizations/my_org/aircrafts/o7flC7jw8jmkOfWo8oyA/flights"
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
  >
    <span
      className="MuiButton-label"
    >
      Alle anzeigen
    </span>
  </a>
</div>
`;

exports[`routes organizations routes aircraft components AircraftDetail renders loading icon if aircraft not loaded 1`] = `
<div
  className="LoadingIcon-container-4"
>
  <div
    className="MuiCircularProgress-root LoadingIcon-icon-5 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;

exports[`routes organizations routes aircraft components AircraftDetail renders loading icon if organization not loaded 1`] = `
<div
  className="LoadingIcon-container-4"
>
  <div
    className="MuiCircularProgress-root LoadingIcon-icon-5 MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
    role="progressbar"
    style={
      Object {
        "height": 40,
        "width": 40,
      }
    }
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
